utilities/assets-proxy-path.js

```diff
@@ -105,15 +105,18 @@ class AssetsProxyPath {
         let body = {
             status: 'PENDING'
         }
+        let milestonesDate = new Date().getTime();
         let response = request('GET', `${this.nodeBaseUrl}/resolve?ids=${id}`);
-        const handler = JSON.parse(response.getBody()).handler_id
+        const handler = JSON.parse(response.getBody()).handler_id;
         while (body.status === 'PENDING') {
-            new Date(new Date().getTime() + 1000);
-            response = request('GET', `${this.nodeBaseUrl}/resolve/result/${handler}`);
-            body = JSON.parse(response.getBody());
+            if((new Date().getTime()) > milestonesDate + 1000){
```

Comment:
Shouldn't we use a sleep (setTimeout) here ? Otherwise the while loop will keep running and it might be cpu intensive

Reply:
I think, there is an issue with this Proxy class not having any async or await or promises in it. introducing the sleep using setTimeout will need the use of promises and will affect the usage of the sdk. I think this is the reason why we are using the sync-request library to make requests in the first place. 

So my solution here seemed the best solution to delay requests, but i agree there is an infinite loop there. 
we need to ask @kotlarmilos form more info about the proxy and whether we rewrite it to use promises instead.